src/provider/EventRecorder.ts

```diff
@@ -147,4 +156,35 @@ export class EventRecorder {
 		this.bundleSequenceId += 1;
 		StorageUtil.saveBundleSequenceId(this.bundleSequenceId);
 	}
+
+	sendEventsInBackground(isWindowClosing: boolean) {
+		if (
+			this.haveFailedEvents &&
+			this.getFailedEventsLength() < NetRequest.KEEP_ALIVE_SIZE_LIMIT
+		) {
+			this.sendFailedEvents();
+			if (isWindowClosing) {
+				StorageUtil.clearFailedEvents();
+			}
+		}
+		if (this.context.configuration.sendMode === SendMode.Batch) {
+			const eventLength = this.getEventsLength();
+			if (eventLength > 0 && eventLength < NetRequest.KEEP_ALIVE_SIZE_LIMIT) {
+				this.flushEvents();
+				if (isWindowClosing) {
+					StorageUtil.clearAllEvents();
```

Comment:
Clear all events is design by default?

Reply:
`StorageUtil.clearAllEvents()` called only when window is closing with network online and not in Firefox, because we can't get the result of the request to set `keepalive`, in this case the browser will do its best to ensure that requests are sent successfully like sendBeacon(sendBeacon also implemented by keepalive, see [Beacon](https://w3c.github.io/beacon/#sendbeacon-method)). In order to avoid repeated sending of events, we choose to delete events in this case.